servers/rendering/renderer_rd/effects/ss_effects.cpp

```diff
@@ -411,12 +411,21 @@ SSEffects::~SSEffects() {
 
 /* SS Downsampler */
 
-void SSEffects::downsample_depth(RID p_depth_buffer, const Vector<RID> &p_depth_mipmaps, bool p_invalidate_uniform_set, Size2i p_full_screen_size, const Projection &p_projection) {
+void SSEffects::downsample_depth(Ref<RenderSceneBuffersRD> p_render_buffers, uint32_t p_view, const Projection &p_projection) {
 	UniformSetCacheRD *uniform_set_cache = UniformSetCacheRD::get_singleton();
 	ERR_FAIL_NULL(uniform_set_cache);
 	MaterialStorage *material_storage = MaterialStorage::get_singleton();
 	ERR_FAIL_NULL(material_storage);
 
+	uint32_t view_count = p_render_buffers->get_view_count();
+	Size2i full_screen_size = p_render_buffers->get_internal_size();
+	Size2i size((full_screen_size.x + 1) / 2, (full_screen_size.y + 1) / 2);
+
+	// Make sure our buffers exist, buffers are automatically cleared if view count or size changes.
+	if (!p_render_buffers->has_texture(RB_SCOPE_SSDS, RB_LINEAR_DEPTH)) {
+		p_render_buffers->create_texture(RB_SCOPE_SSDS, RB_LINEAR_DEPTH, RD::DATA_FORMAT_R16_SFLOAT, RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT, RD::TEXTURE_SAMPLES_1, size, view_count * 4, 6);
```

Comment:
Previously we only created 5 mipmaps, but here we create 6. What is the 6th mipmap used for?

Reply:
Ah, I did this because in the old code the last mip was always set to the 5th mip and in theory that would double with the previous mip we selected and I figured that may be a bug. But this mip is only used if `use_full_mips` is set, and that is only set if `use_half_size` isn't, so it would not run into the 6th.

I'll fix that up :)

Reply:
Ok done.